kyo-prelude/shared/src/main/scala/kyo2/kernel/Safepoint.scala

```diff
@@ -92,6 +96,38 @@ object Safepoint:
         immediate(p)(loop(v))
     end propagating
 
+    private[kyo2] def ensure[A, S](f: => Unit)(v: => A < S)(using safepoint: Safepoint, _frame: Frame): A < S =
+        // ensures the function is called once even if an
+        // interceptor executes it multiple times
```

Comment:
This behavior is necessary because `f` can be executed multiple times depending on the execution path. For example the function can be evaluated at the end of the computation normally but then an exception is thrown and the catch in `eval` attempts to execute it again. `IOTask` also has some performance optimizations that can end up invoking the function multiple times.

Reply:
Very clever! Perhaps we should name `def eval` distinctly from the function which is a few lines below `ensure`?